pages/blog/miniseries-part_1.md

```diff
@@ -0,0 +1,227 @@
+
+My side project at that time was a [Rust](https://rust.facepunch.com/) game server plugin which collected in game events, such as when a player farms resources, kills another player, loots a container, etc, and send them to a backend. Later these could be extracted by an API to display the players progression and detailed account on what you did on the game server. 
+
+Initially I used OpenAPI to describe the API and the great thing was their tooling allowed me generate clients and servers in different languages which really speed up the implementation process.
+
+I soon encountered a use case that required me to push data to the game server, and solving this with REST was possible, but cumbersome. So I started exploring different alternatives, in terms of event driven architecture, however, none could be described using OpenAPI or offered tooling, so I had to find alternatives. 
+
+That was when I vaguely remembered a meeting where AsyncAPI was mentioned. Around that time, at work, we began to switch from a custom socket protocol to [NATS](https://nats.io/), and at the same time they spend some time figuring out how to mainstream the process and document the API's, and this is where they found and adopted AsyncAPI. So I started to use AsyncAPI in my project, and sparked my first ever encounter with open source, but that is a story for another time, maybe.
+
+So this blog post is a dedication to that experience, showcasing how I use AsyncAPI to document and generate code to speed up the development process, and maybe spark your interest in building the best tooling possible.
+
+# To that end
+
+Explaining something is always better with actual examples, therefore I will be creating a little system to show you how code generation can help the development process. 
+
+![General setup](/img/posts/jonaslagoni-miniseries-part1/blog-miniseries-general-setup.webp)
+
+I will be creating a system of two applications, a **game server** and a **processor** using a micro-service architecture with no public facing API. How a player interact with the **game server** could be through a phone, a computer, Xbox or PlayStation, for us this is out of scope for this project, we only care about the interaction between the **game server** and the **processor**. The round dot between some broker and the game server represent how others may grab/interact with the application, ergo its API. The same thing is for the processor.
+
+The main point of the **game server** will be to simulate the interaction of the player. This means it will simulate the following events: when players join the game server, pick up items in game, uses the chat, hit one another, and eventually disconnect.
```

Comment:
what `simulate` means here really? like it will pretend, play, simulate the Player or? if yes, then why? why Player is then on the diagram if it is anyway simulated by the game server

Reply:
I added the player to the image since it made sense to me still displaying them, regardless of whether the game server simulates the players or not. Simulating these events simply means it will at random intervals simulate players is joining the server, picking up some items, etc, and eventually disconnecting.

I will clarify it some more.